src/pages/quiz-list-page/quiz-list-page.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useState } from "react";
+import { WelcomeMessage } from "../../common/components/homepage/welcome-message/welcome-message";
+
+import "../no-quizzes-page/no-quizzes.scss";
+import "./quiz-list-page.scss";
+import clock_icon from "../../common/assets/clock-icon.svg";
+import file_icon from "../../common/assets/paper-icon.svg";
+import edit_icon from "../../common/assets/edit-icon.svg";
+import sword_icon from "../../common/assets/sword_icon.png";
+
+import { Button } from "primereact/button";
+import { Card } from "primereact/card";
+import { useNavigate, createSearchParams } from "react-router-dom";
+import { LoginStore } from "../../features/authentication/login/store/LoginStore";
+
+import { collection, getDocs } from "firebase/firestore";
+import { db } from "../../config/firebase-config";
+import { QuizStore } from "../../features/create-quiz/store/CreateQuizStore";
+
+interface QuizListProps {}
+
+export const QuizList: React.FC<QuizListProps> = () => {
+  const navigate = useNavigate();
+
+  const [quizzes, setQuizzes] = useState<{ [key: string]: any }[]>([]);
+  const quizzesCollectionRef = collection(db, "quizzes");
+
+  const { user } = LoginStore.login;
+
+  const convertTimeStampToDate = (timeStamp: number) => {
```

Comment:
This function is generic so it could be extracted globally into an `util` file so it could be used anywhere needed.